src/Confluent.SchemaRegistry.Serdes.Avro/SpecificSerializerImpl.cs

```diff
@@ -115,39 +169,61 @@ internal class SpecificSerializerImpl<T> : IAvroSerializerImpl<T>
                 );
             }
 
-            avroWriter = new SpecificWriter<T>(writerSchema);
-            writerSchemaString = writerSchema.ToString();
+            specificSerializerSchemaData.AvroWriter = new SpecificWriter<T>(specificSerializerSchemaData.WriterSchema);
+            specificSerializerSchemaData.WriterSchemaString = specificSerializerSchemaData.WriterSchema.ToString();
+            return specificSerializerSchemaData;
         }
 
         public async Task<byte[]> Serialize(string topic, T data, bool isKey)
         {
             try
-            {
+            {   
+                SpecificSerializerSchemaData currentSchemaData;
                 await serializeMutex.WaitAsync().ConfigureAwait(continueOnCapturedContext: false);
                 try
                 {
+                    if (_isMultipleSchemas)
+                    {
+                        var key = data.GetType();
+                        if (!schemaData.TryGetValue(key, out currentSchemaData))
+                        {
+                            currentSchemaData = ExtractSchemaData(key);
+                            schemaData[key] = currentSchemaData;
+                        }
+                    }
+                    else
+                    {
+                        currentSchemaData = schemaData.Values.First();
+                    }
+                    
+                    
                     string fullname = null;
-                    if (data is ISpecificRecord && ((ISpecificRecord)data).Schema is Avro.RecordSchema)
+                    if (data is ISpecificRecord && ((ISpecificRecord) data).Schema is Avro.RecordSchema)
```

Comment:
keep whitespace as is unless good reason to change (allows for more useful git blame)